Adding vim, VS Code and some tmux shortcuts

ratnadeepb opened the pull request:
Adding a small guide to vim along with a config file and added a few tmux options

ratnadeepb committed:
Adding vim and a different tmux config

ratnadeepb committed:
removing complex options

ratnadeepb committed:
removing complex options

ratnadeepb committed:
adding vim folder and readme

ratnadeepb committed:
adding buffer mgmt

ratnadeepb committed:
adding tmux shortcuts

ratnadeepb committed:
pattern

ratnadeepb committed:
minor editing

ratnadeepb committed:
minor editing

ratnadeepb committed:
Adding for VS Code

ratnadeepb committed:
Editing

ratnadeepb committed:
Editing

ratnadeepb committed:
Editing

ratnadeepb commented:
Adding VS Code startup README

gparmer commented on README.md:
```diff
@@ -8,3 +8,6 @@ Please do pull requests for additional tools.
 If you want to use a tool, copy the dotfile(s) for it to your home directory (unless otherwise noted).
 The `README.md` in each tool's directory should include an overview of the tool, an overview of how to use it (for beginners), and a set of links for people to learn more.
+
+## Regarding VIM config
```
Please remove this.

ratnadeepb replied:
Done in commit e0f7e91

gparmer commented on tmux/README.md:
```diff
@@ -42,3 +42,16 @@ The next keypresses you make will be interpreted as a command to `tmux`, which a
+- Open a new session
+    - tmux new
+    - tmux new dev -> named session called "dev"
+- Tmux command mode (Ctrl + b)
```
Doesn't this conflict with the actual config that redefines the control code?

ratnadeepb replied:
Could you clarify this point please. I have so far had no issues with named sessions.

gparmer replied:
The tmux config that is in the tree uses a different control sequence, not `cntl + b`, so including documentation that implies it is `cntl + b` is not quite right.

ratnadeepb replied:
In the tmux config I use `cntl+a` is the secondary `<prefix>` but `cntl+b` still works.

ratnadeepb replied:
I changed it to `cntl + t`. I had assumed, erroneously, that since unbind was not used `cntl + b` would still work.

ratnadeepb replied:
Commit e288d5c

gparmer commented on VS_Code/README.md:
```diff
@@ -0,0 +1,73 @@
+# Visual Studio Code
```
Sorry: I just merge @bushidocodes work on vscode. You can add/modify that.

Sorry I'm causing you issues here!

ratnadeepb replied:
No worries. That was a brain fade on my part. I will make the necessary changes.

ratnadeepb replied:
In commit bd616f5

gparmer commented on vim/README.md:
```diff
@@ -0,0 +1,123 @@
+# VIM
```
This all looks great. I want to merge the vim stuff as-is.

ratnadeepb committed:
adding vim dotfile

ratnadeepb committed:
Removing VS Code from this pull request

Will make a new pull request later to add it to the existing directory

ratnadeepb committed:
removing comment from main README

ratnadeepb committed:
Adding clang formatting and adding to vs code readme

Added dotfile for clang format. Also added some notes on setting up
remote development with VS Code

ratnadeepb committed:
minor edits

ratnadeepb commented:
Incorporated all suggestions

ratnadeepb committed:
Changed the command mode
